src/main/java/reposense/commits/CommitResultAggregator.java

```diff
@@ -122,16 +127,64 @@ private static void addDailyContributionForNewDate(
         authorDailyContributions.add(new AuthorDailyContribution(date));
     }
 
-    private static Date getStartOfDate(Date current) {
-        Calendar cal = Calendar.getInstance();
-        cal.setTime(current);
+    /**
+     * Get the starting point of the {@code current} date with respect to the {@code zoneId} timezone.
+     */
+    private static Date getStartOfDate(Date current, ZoneId zoneId) {
+        if (current.equals(SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE)) {
+            return current;
+        }
+
+        Instant now = Instant.now();
```

Comment:
Since line 138 until line 142 is being re-used to get the `systemRawOffset - zoneRawOffset` in multiple methods, perhaps can refactor to a separate method?